govcd/orgvdcnetwork_test.go

```diff
@@ -31,3 +32,176 @@ func (vcd *TestVCD) Test_NetRefresh(check *C) {
+			IPScopes: &types.IPScopes{
+				IPScope: types.IPScope{
+					IsInherited: false,
+					Gateway:     "10.10.102.1",
+					Netmask:     "255.255.255.0",
+					IPRanges: &types.IPRanges{
+						IPRange: []*types.IPRange{
+							&types.IPRange{
+								StartAddress: "10.10.102.2",
+								EndAddress:   "10.10.102.100"},
+						},
+					},
+				},
+			},
+			BackwardCompatibilityMode: true,
+		},
+		EdgeGateway: &types.Reference{
+			HREF: edgeGateway.EdgeGateway.HREF,
+			ID:   edgeGateway.EdgeGateway.ID,
+			Name: edgeGateway.EdgeGateway.Name,
+			Type: edgeGateway.EdgeGateway.Type,
+		},
+		IsShared: false,
+	}
+
+	LogNetwork(networkConfig)
+	err = vcd.vdc.CreateOrgVDCNetwork(&networkConfig)
+	if err != nil {
+		fmt.Printf("error creating Network <%s>: %s\n", networkName, err)
+	}
+	check.Assert(err, IsNil)
+	AddToCleanupList(TestCreateOrgVdcNetworkEGW,
+		"network",
+		vcd.org.Org.Name+"|"+vcd.vdc.Vdc.Name,
+		"Test_CreateOrgVdcNetworkEGW")
+}
+
+// Tests the creation of an isolated Org VDC network
+func (vcd *TestVCD) Test_CreateOrgVdcNetworkIso(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	networkName := TestCreateOrgVdcNetworkIso
+
+	err := RemoveOrgVdcNetworkIfExists(vcd.vdc, networkName)
+	if err != nil {
+		check.Skip(fmt.Sprintf("Error deleting network : %s", err))
+	}
+
+	var networkConfig = types.OrgVDCNetwork{
+		Xmlns: "http://www.vmware.com/vcloud/v1.5",
+		Name:  networkName,
+		// Description: "Created by govcd tests",
+		Configuration: &types.NetworkConfiguration{
+			FenceMode: "isolated",
+			/*One of:
+				bridged (connected directly to the ParentNetwork),
+			  isolated (not connected to any other network),
+			  natRouted (connected to the ParentNetwork via a NAT service)
+			  https://code.vmware.com/apis/287/vcloud#/doc/doc/types/OrgVdcNetworkType.html
+			*/
+			IPScopes: &types.IPScopes{
+				IPScope: types.IPScope{
+					IsInherited: false,
+					Gateway:     "192.168.2.1",
+					Netmask:     "255.255.255.0",
+					DNS1:        "192.168.2.1",
+					DNS2:        "",
+					DNSSuffix:   "",
+					IPRanges: &types.IPRanges{
+						IPRange: []*types.IPRange{
+							&types.IPRange{
+								StartAddress: "192.168.2.2",
+								EndAddress:   "192.168.2.100"},
+						},
+					},
+				},
+			},
+			BackwardCompatibilityMode: true,
+		},
+		IsShared: false,
+	}
+	LogNetwork(networkConfig)
+	err = vcd.vdc.CreateOrgVDCNetwork(&networkConfig)
+	if err != nil {
+		fmt.Printf("error creating Network <%s>: %s\n", networkName, err)
+	}
+	check.Assert(err, IsNil)
+	AddToCleanupList(TestCreateOrgVdcNetworkIso,
+		"network",
+		vcd.org.Org.Name+"|"+vcd.vdc.Vdc.Name,
+		"Test_CreateOrgVdcNetworkIso")
+}
+
+// Tests the creation of a Org VDC network connected to an external network
+func (vcd *TestVCD) Test_CreateOrgVdcNetworkDirect(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	networkName := TestCreateOrgVdcNetworkDirect
+
+	err := RemoveOrgVdcNetworkIfExists(vcd.vdc, networkName)
```

Comment:
I see some parts are good option for removing repeatable code.

Reply:
I have already squeezed the non-meaningful operations into functions. Since the file contains three different tests, I prefer leaving the details spelled out, as the tests can fail in different ways.